cgra/CgraTemplateRTL.py

```diff
@@ -161,6 +167,38 @@ def construct(s, DataType, PredicateType, CtrlPktType, CgraPayloadType,
         dstTileIndex = link.dstTile.getIndex(TileList)
         s.tile[srcTileIndex].send_data[link.srcPort] //= s.tile[dstTileIndex].recv_data[link.dstPort]
 
+    if is_multi_cgra: 
+      for row in range(height):
+        for col in range(width):
+          tile_id = row * width + col
+          if row == height - 1:
+            # Only connect if the port is valid
+            if PORT_NORTH not in TileList[tile_id].getInvalidOutPorts():

```

Comment:
Shouldn't we connect
```
s.send_data_on_boundary_north[col] // 0
```
if `PORT_NORTH in TileList[tile_id].getInvalidOutPorts():`?